compiler/rustc_middle/src/ty/assoc.rs

```diff
@@ -43,7 +43,6 @@ impl AssocItemContainer {
 #[derive(Copy, Clone, Debug, PartialEq, HashStable, Eq, Hash)]
 pub struct AssocItem {
     pub def_id: DefId,
-    #[stable_hasher(project(name))]
     pub ident: Ident,
```

Comment:
IIRC, this field is used for name resolution. So it really only needs the `Symbol` and the `SyntaxContext`. In case of diagnostics, the span can be recovered with the `def_ident_span` query.
Should we introduce a `struct HygienicIdent { name: Symbol, ctxt: SyntaxContext }` for this purpose? This could avoid part of the regression due to moved code.

Reply:
I came to the same conclusion about the `Span`. I'm working on a branch locally where I've switched to just storing a `Symbol`, and using `def_ident_span` to reconstruct the `Ident` in the few cases where it's actually needed.